File: src/main/java/com/wxl/dyttcrawler/urlhandler/UrlFilter.java
package com.wxl.dyttcrawler.urlhandler;

import java.net.URL;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Create by wuxingle on 2020/5/16
 * url过滤
 */
public interface UrlFilter {


    /**
     * 是否匹配
     */
    boolean filter(URL url);


    /**
     * 提取匹配的url
     */
    default List<URL> filter(List<URL> urls) {
        return urls.stream()
                .filter(this::filter)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/wxl/dyttcrawler/urlhandler/UrlFilter.kt
package com.wxl.dyttcrawler.urlhandler

import java.net.URL

/**
 * Create by wuxingle on 2021/10/11
 * url过滤
 */
fun interface UrlFilter {

    /**
     * 是否匹配
     */
    fun match(url: URL): Boolean

    /**
     * 提取匹配的url
     */
    fun filter(urls: List<URL>): List<URL> {
        return urls.filter { match(it) }
    }

}
